### ISSUE TITLE
Better Deconstruction

### ISSUE THREAD

**Issue by ComputerOverlord** [TeamPorcupine/ProjectPorcupine#180]
The idea is you deconstruct (or mine) stuff and get some (if not most) of the materials back. 
This might be great for Quills tutorial series.

**Comment by satch3l** [comment]
This is probably not priority for the moment. But I agree with you, deconstruction should yield something like 50-75% of total construction cost.

**Comment by ComputerOverlord** [comment]
That would also allow for mining asteroids...

**Comment by vogonistic** [comment]
I think this might be a perfectly reasonable time to work on this.

**Comment by GamerInspired** [comment]
I agree with @vogonistic this is a good time to work on this.

**Comment by kd7uiy** [comment]
This should also cover the issue mentioned in #286. With that, I will close that one. Just make sure we add deconstruction to the queue.

**Comment by kd7uiy** [comment]
Okay, so this has been opened for quite a while. I think I'm going to make this my next project, if no one objects.

**Comment by TomMalbran** [comment]
@kd7uiy Someone already started it (check PR linked above). We should first check if that person will finish it or something. But it seems that he will not. You could work on top of his PR.

**Comment by sboigelot** [comment]
@kd7uiy @TomMalbran I'm closing the previous PR on this issue as there is no sign of life since 16 days on it.

**Comment by TomMalbran** [comment]
That makes sense. We should give closure to old PR before starting to work on them again. We can still work on top of that PR, if it was good enough.

**Comment by sboigelot** [comment]
Hard to do as the code evolve a fast, a 2 week old PR touching core component and never updated, we are better of starting from zero.

**Comment by TomMalbran** [comment]
Yes that is true. But some modules haven't changed much, and if not the idea can be copied. Haven't really checked the PR so I don't really know.

**Comment by kd7uiy** [comment]
I looked at the old PR, it seems that some of the systems that are in place might not work well with it. But I can always pull the code from that PR, and I might give it a test run, and use at least a portion of it.

**Comment by sboigelot** [comment]
@kd7uiy take care about that old PR, a lot of things looked good but it also touch the pathfinding in private area :)

**Comment by vogonistic** [comment]
That PR was held up partially because of how the adjacent job was solved. The PR on pathfinding refactoring I made handles that, so it should be easier after that.
